### DISCUSSION TITLE
Build issues

### DISCUSSION THREAD

**Discussion by zlgonzalez** [jrincayc/ucblogo-code#168] [General]
Hi,
  I'm on Ubuntu 20.04, and I am trying to build UCB Logo. I followed the instructions, but got an issue when invoking ./configure:

checking enable_x11... no
checking enable_docs... yes
./configure: line 5328: AX_CFLAGS_WARN_ALL: command not found
./configure: line 5329: syntax error near unexpected token `-Wno-write-strings,'
./configure: line 5329: `AX_CHECK_COMPILE_FLAG(-Wno-write-strings,'

Thoughts?

Thanks,
Ron

**Comment by jrincayc** [comment]
Hm, did you start from the git version or a tar.gz?

**Reply by zlgonzalez** [reply]
I tried both and had issues for both.

**Reply by zlgonzalez** [reply]
autoconf (GNU Autoconf) 2.71

**Comment by dmalec** [comment]
Interesting, I ran into the same error when compiling (from the git version) on Haiku; but, had thought it was limited to something specific about that OS. Given I've only hit this on Haiku, I worked around it by commenting out all added compiler flags using `dnl`. Essentialy, this block of `configure.ac`:
```
dnl Enable Compiler Warnings
AX_CFLAGS_WARN_ALL
dnl Disable a warning with many false positives; perhaps rethink later
AX_CHECK_COMPILE_FLAG([-Wno-write-strings],
[CFLAGS="$CFLAGS -Wno-write-strings"
CXXFLAGS="$CXXFLAGS -Wno-write-strings"])
dnl Disable voluminous cloud of rather harmless warnings
AX_CHECK_COMPILE_FLAG([-Wno-unused-variable],
[CFLAGS="$CFLAGS -Wno-unused-variable"
CXXFLAGS="$CXXFLAGS -Wno-unused-variable"])
```

becomes this:
```
dnl Enable Compiler Warnings
dnl AX_CFLAGS_WARN_ALL
dnl Disable a warning with many false positives; perhaps rethink later
dnl AX_CHECK_COMPILE_FLAG([-Wno-write-strings],
dnl [CFLAGS="$CFLAGS -Wno-write-strings"
dnl CXXFLAGS="$CXXFLAGS -Wno-write-strings"])
dnl Disable voluminous cloud of rather harmless warnings
dnl AX_CHECK_COMPILE_FLAG([-Wno-unused-variable],
dnl [CFLAGS="$CFLAGS -Wno-unused-variable"
dnl CXXFLAGS="$CXXFLAGS -Wno-unused-variable"])
```

That said, seeing it is happening on Ubuntu made me go back over my work. I think the core issue is a missing package that contains the macros in question. On Haiku, this can be installed by adding the `autoconf_archive` package via HaikuDepot and I can now uncomment those lines and run `./compile` without error.

On Ubuntu, I suspect this can potentially be resolved by installing `autoconf-archive`; however, I'm not sure I'll be able to replicate the 20.04 environment until maybe the weekend, so there's a bit of _caveat emptor_ in this suggestion.

**Reply by zlgonzalez** [reply]
Thanks I'll give that suggestion a try. It is also failing for me on Mac after I brew installed autoconf automake and libtool. After I installed it, it got me past that error on to a new compile error in make. I will create a new discussion for that unless you prefer we continue this discussion here.

**Reply by zlgonzalez** [reply]
Install autoconf-archive indeed fixed the problem on Ubuntu. So now the issue I am running into is with the build in terms of linking tputs, tgoto, etc. Will see what other library needs to be installed.

**Reply by dmalec** [reply]
Ah, just to verify, are you working on compiling a non-wxWidgets build? I may be off base; but, I think `tputs`, `tgoto`, etc. come into play when not linking against wxWidgets.

**Reply by zlgonzalez** [reply]
I can use wxwidgets. Is there a page that covers all the required dependencies that I may have missed?

**Reply by dmalec** [reply]
This Wiki page is the current capture:
https://github.com/jrincayc/ucblogo-code/wiki/package_creation

I am realizing I should update that for our recent move to wxWidgets 3.2.x (wxWidgets 3.0.5 should still work fine, it's more that we can compile and run with either now and have moved the compiled OSX and Windows builds to 3.2.2.1).

The CI & CD YAMLs are also a place you can verify against (though, I say that while acknowledging that the information there is machine friendly more than human friendly):
* https://github.com/jrincayc/ucblogo-code/blob/master/.github/workflows/ci.yml
* https://github.com/jrincayc/ucblogo-code/blob/master/.github/workflows/cd.yaml

On Ubuntu 22.04 w/ wxWidgets 3.0.5 this is what I am doing for build dependencies:
```
sudo apt-get install \
        autoconf-archive \
        libgtk-3-dev \
        libwxgtk3.0-gtk3-dev \
        libnotify-dev
```

On Ubuntu 22.0.4 w/ wxWidgets 3.2 this is what I'm doing:
```
sudo apt-key adv --fetch-keys http://repos.codelite.org/CodeLite.asc
sudo apt-add-repository 'deb https://repos.codelite.org/wx3.2.2.1/ubuntu/ jammy universe'

sudo apt-get install \
        autoconf-archive \
        libgtk-3-dev \
        libwxgtk3.2unofficial-dev \
        libnotify-dev
```

**Reply by zlgonzalez** [reply]
Thanks! Will give this a whirl.

**Reply by zlgonzalez** [reply]
Hi,
  The first one seems to work, but the second one with 3.2 says that libwxgtk3.2unofficial-dev doesn't exist.

Thanks,
Ron

**Comment by zlgonzalez** [comment]
So I was able to get it finally building and running without adding all these dependencies in my Linux local dev environment by creating a Docker image (adding all the libraries from wxwidgets scared me :)).

Anyway, I would love to try building with wxwidgets 3.2 if you can respond to what the actual library name should be instead of libwxgtk3.2unofficial-dev.

**Reply by dmalec** [reply]
On Ubuntu 20.04, I believe the library is named `libwxgtk3.2unofficial-dev`. I set up an environment and was able to do the following:
```
sudo apt-key adv --fetch-keys http://repos.codelite.org/CodeLite.asc

sudo apt-add-repository 'deb https://repos.codelite.org/wx3.2.2.1/ubuntu/ focal universe'

sudo apt-get install libwxgtk3.2unofficial-dev
```

Which produced this output:
```
Reading package lists... Done
Building dependency tree       
Reading state information... Done

The following additional packages will be installed:
  libegl-dev libgl-dev libgl1-mesa-dev libgles-dev libgles1 libglu1-mesa-dev libglvnd-dev libglx-dev libopengl-dev
  libpthread-stubs0-dev libwxbase3.2-0-unofficial libwxbase3.2unofficial-dev libwxgtk3.2-0-unofficial libx11-dev
  libxau-dev libxcb1-dev libxdmcp-dev wx-common wx3.2-headers x11proto-core-dev x11proto-dev xorg-sgml-doctools
  xtrans-dev

Suggested packages:
  gettext libx11-doc libxcb-doc

The following NEW packages will be installed:
  libegl-dev libgl-dev libgl1-mesa-dev libgles-dev libgles1 libglu1-mesa-dev libglvnd-dev libglx-dev libopengl-dev
  libpthread-stubs0-dev libwxbase3.2-0-unofficial libwxbase3.2unofficial-dev libwxgtk3.2-0-unofficial
  libwxgtk3.2unofficial-dev libx11-dev libxau-dev libxcb1-dev libxdmcp-dev wx-common wx3.2-headers x11proto-core-dev
  x11proto-dev xorg-sgml-doctools xtrans-dev

0 upgraded, 24 newly installed, 0 to remove and 4 not upgraded.

Need to get 9348 kB of archives.

After this operation, 41.4 MB of additional disk space will be used.
```
